applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -989,12 +999,15 @@ static void update_key(const struct hid_keymap *map, s16_t value)
 {
 	u8_t report_id = map->report_id;
 
-	size_t pos = report_index[report_id];
-	__ASSERT_NO_MSG(pos < INPUT_REPORT_COUNT);
+	struct report_data *rd = get_report_data(report_id);
+
+	bool connected = false;
 
-	struct report_data *rd = &state.report_data[pos];
-	bool connected = state.selected &&
-		(state.selected->state[pos].state != STATE_DISCONNECTED);
+	if (state.selected) {
+		struct report_state *rs = get_report_state(state.selected, report_id);
```

Comment:
`__ASSERT_NO_MSG(rs != NULL)`

Reply:
we cannot move the assert there as NULL is a valid return used in one case

Reply:
Ok, let's leave it as it is.